### DISCUSSION TITLE
Rehydratable Component needs improvements

### DISCUSSION THREAD

**Discussion by AJCM-git** [space-wizards/space-station-14#4050] [General]
- It doesn't handle injecting solutions with a syringe (SolutionComponent does)
- It should support spraying the right reagents over the entities with the component and activating it
- It should activate when you put the rehydratable entity over a puddle with the right reagents

**Comment by 20kdc** [comment]
+ Syringes won't inject unless something's there to inject into, etc. `RehydratableComponent` does have `IReagentReaction` for `ReagentReactInjection` and `ReagentReactTouch` so if there's an issue, it shouldn't be in that component.
+ Spraying... I don't know why this doesn't already work, something about collision masks maybe
+ Puddles... same
